pkg/kubelet/eviction/eviction_manager_test.go

```diff
@@ -59,16 +60,16 @@ type mockPodKiller struct {
 }
 
 // killPodNow records the pod that was killed
-func (m *mockPodKiller) killPodNow(pod *v1.Pod, evict bool, gracePeriodOverride *int64, semaphore chan bool, statusFn func(*v1.PodStatus)) error {
-	m.killPodNowNoSemaphoreRelease(pod, evict, gracePeriodOverride, semaphore, statusFn)
-	if semaphore != nil {
-		<-semaphore
+func (m *mockPodKiller) killPodNow(pod *v1.Pod, evict bool, gracePeriodOverride *int64, lock *sync.Mutex, statusFn func(*v1.PodStatus)) error {
+	_ = m.killPodNowLongShutdown(pod, evict, gracePeriodOverride, lock, statusFn)
+	if lock != nil {
+		lock.Unlock()
 	}
 	return nil
 }
 
-// killPodNowNoSemaphoreRelease records the pod that was killed, and does not release the semaphore, simulating a long pod shutdown
-func (m *mockPodKiller) killPodNowNoSemaphoreRelease(pod *v1.Pod, evict bool, gracePeriodOverride *int64, semaphore chan bool, statusFn func(*v1.PodStatus)) error {
+// killPodNowLongShutdown records the pod that was killed, and does not unlock the lock, simulating a long pod shutdown
+func (m *mockPodKiller) killPodNowLongShutdown(pod *v1.Pod, evict bool, gracePeriodOverride *int64, lock *sync.Mutex, statusFn func(*v1.PodStatus)) error {
```

Comment:
In the code itself, `eviction.NewManager()` is always being called with `killPodFunc: killPodNowAsync(klet.podWorkers, kubeDeps.Recorder)`, so I wonder if it's necessary.

I think that in the test you should provide the same function (i.e. `killPodNowAsync`) as an argument for the manager instead.

WDYT? Am I missing something?

Reply:
Hmmm, so that will mean re-working the tests so that they don't have this mockPodKiller and instead have mocked out podworkers that record the same info (what was evicted, grace period, etc). Is that what you mean?

Reply:
hmmm, PodWorkers are in the kubelet package, which causes a circular dependency. I can refactor all of this, but that would be a huge change